### PR TITLE
[PACKAGING] Push docker images with the architecture in the version

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

Support multiplatform docker images:

- with the `-amd64` in the versioning
- with the `-arm64` in the versioning
- Default docker image for amd64 arch only. <--- This should eventually be removed once the E2E supports the arch in the versioning.

Enable the E2E testing

Support the above implementation as part of the existing `Packaging` pipeline and the `packaging` stage in the pipeline 2.0 

## Why is it important?

Avoid the override of docker images that are supporting AMD and ARM

### CODE REVIEW

File: .ci/packaging.groovy [hunk 1]
```diff
@@ -214,6 +216,7 @@ pipeline {
               }
             }
             stages {
+              // This should always after the linux/amd64 packaging to support the multiplatform docker registry
```
Comment (v1v) [inline .ci/packaging.groovy:219]:
```suggestion
```

File: .ci/packaging.groovy [hunk 2]
```diff
@@ -247,15 +251,13 @@ pipeline {
             }
           }
         }
-        /*
```
Comment (v1v) [inline .ci/packaging.groovy:250]: Let's enable the E2E :)

**Review by kuisathaverat — APPROVED** [review]

File: .ci/packaging.groovy (hunk 2 above)
Comment (mdelapenya) [reply .ci/packaging.groovy:250]: How about having them disabled until we use the new name-tag strategy for multiplatform?

Comment (mdelapenya) [reply .ci/packaging.groovy:250]: Ups, I saw the comment for maintaining the default tag meanwhile. LGTM!

**Review by mdelapenya — APPROVED** [review]

**Review by cachedout — APPROVED** [review]

File: .ci/packaging.groovy [hunk 3]
```diff
@@ -309,14 +321,22 @@ def tagAndPush(beatName){
...
-    doTagAndPush(beatName, variant, libbetaVer, "${env.GIT_BASE_COMMIT}")
-
-    if (!isPR() && aliasVersion != "") {
-      doTagAndPush(beatName, variant, libbetaVer, aliasVersion)
+    tags.each { tag ->
+      // TODO:
+      // For backward compatibility let's ensure we tag only for amd64, then E2E can benefit from until
+      // they support the versioning with the architecture
+      if ("${arch}" == "amd64") {
+        doTagAndPush(beatName: beatName, variant: variant, sourceTag: libbetaVer, targetTag: "${tag}")
```
Comment (mdelapenya) [inline .ci/packaging.groovy:337]: Addresses the change in https://github.com/elastic/e2e-testing/pull/787

**Pull request merged**